src/sparseml/transformers/utils/helpers.py

```diff
@@ -224,3 +209,132 @@ def resolve_sequence_length(config: AutoConfig) -> int:
+    recipe: Union[str, Path, None], model_path: Union[str, Path]
+) -> Union[str, Path, None]:
+    """
+    Resolve the recipe to apply to the model.
+    :param recipe: the recipe to apply to the model.
+        It can be one of the following:
+        - None
+        - a path to the recipe file (string or Path)
+        - name of the recipe file (e.g. "recipe.yaml")
+            (assumed to be stored in the model_path instead
+            of RECIPE_NAME)
+        - a string containing the recipe
+
+        If recipe is None, the function will try to
+        find the recipe in the model_path if model_path
+        is a local path. If model_path is a huggingface
+        model_id, the function will try to download the
+        recipe from huggingface.co. If model_path is a
+        sparsezoo stub, the function will try to download
+        the recipe from sparsezoo
+
+    :param model_path: the path to the model to load
+    :return: the resolved recipe
+    """
+
+    if recipe is None:
+        return _infer_recipe_from_model_path(model_path)
+
+    elif os.path.isfile(recipe):
+        # recipe is a path to a recipe file
+        return _resolve_recipe_file(recipe, model_path)
+
+    elif os.path.isfile(os.path.join(model_path, recipe)):
+        # recipe is a name of a recipe file
+        recipe = os.path.join(model_path, recipe)
+        return _resolve_recipe_file(recipe, model_path)
+
+    elif isinstance(recipe, str):
+        # recipe is a string containing the recipe
+        _LOGGER.debug(
+            "Applying the recipe string directly to the model, without "
+            "checking for a potential existing recipe in the model_path."
+        )
+        return recipe
+
+    _LOGGER.info(
+        "No recipe requested and no default recipe "
+        f"found in {model_path}. Skipping recipe resolution."
+    )
+    return None
+
+
+def _infer_recipe_from_model_path(model_path: Union[str, Path]) -> Optional[Union[str]]:
+    if isinstance(model_path, Path) or os.path.isdir(model_path):
+        # model_path is a local path
+        model_path = (
+            model_path.as_posix() if isinstance(model_path, Path) else model_path
+        )
+        recipe = os.path.join(model_path, RECIPE_NAME)
+        if os.path.isfile(recipe):
+            # if recipe is an actual file -> use it
+            return recipe
+        return None
+
+    elif model_path.startswith("zoo:"):
+        # model path is a sparsezoo stub
+        recipe = Model(model_path).recipes.default.path
+        return recipe
+
+    model_id = os.path.join(HUGGINGFACE_CO_URL_HOME, model_path)
+    request = requests.get(model_id)
+    if request.status_code == 200:
+        # model path is a valid huggingface model id
+        _LOGGER.info(
+            "model_path is a huggingface model id. "
+            f"Attempting to download recipe from {HUGGINGFACE_CO_URL_HOME}"
+        )
+        recipe = hf_hub_download(repo_id=model_path, filename=RECIPE_NAME)
```

Comment:
need to protect against an HF model that does not have a recipe - should return None